### PR TITLE
Fix not honouring the non Proxy ports

### PR DESCRIPTION
**Pull request by RAVEENSR**
## Purpose
This PR fixes issue wso2/product-apim#9465

### CODE REVIEW

File: components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java [hunk 1]
```diff
@@ -594,12 +597,18 @@ private static boolean isInvalidHeader(String header) {
...
 
         ProxyHost proxyConfig = null;
 
-        String proxyHost = System.getProperty("http.proxyHost");
-        String proxyPortStr = System.getProperty("http.proxyPort");
+        String proxyHost = System.getProperty(HTTP_PROXY_HOST);
+        String proxyPortStr = System.getProperty(HTTP_PROXY_PORT);
+        String nonProxyHosts = System.getProperty(HTTP_NON_PROXY_HOSTS);
+
+
```
Comment (1akshitha) [inline components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java:608]: remove the unnecessary line

File: components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java [hunk 2]
```diff
@@ -621,4 +630,193 @@ private ProxyHost getProxyConfig() {
...
+        // All characters in the string are used. Check if only '*'s are left
+        // in the pattern. If so, we succeeded. Otherwise failure.
+        for (int i = patIdxStart; i <= patIdxEnd; i++) {
+            if (patArr[i] != '*') {
+                return false;
+            }
+        }
+        return true;
+    }
+
```
Comment (1akshitha) [inline components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java:821]: empty line

File: components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java [hunk 3]
```diff
@@ -621,4 +630,193 @@ private ProxyHost getProxyConfig() {
...
+     *
+     * @param pattern         the (non-null) pattern to match against
+     * @param str             the (non-null) string that must be matched against the
+     *                        pattern
+     * @param isCaseSensitive
+     * @return <code>true</code> when the string matches against the pattern,
+     *         <code>false</code> otherwise.
+     */
+    private static boolean match(String pattern, String str,
+                                 boolean isCaseSensitive) {
```
Comment (1akshitha) [inline components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java:673]: shall we get this to the previous line?

File: components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java (hunk 1 above)
Comment (RAVEENSR) [reply components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java]: Fixed in af3ac874ea63a07e139fdd85b36eeeed3d15e410

File: components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java (hunk 2 above)
Comment (RAVEENSR) [reply components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java]: Fixed in af3ac874ea63a07e139fdd85b36eeeed3d15e410

File: components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java (hunk 3 above)
Comment (RAVEENSR) [reply components/hostobjects/org.jaggeryjs.hostobjects.xhr/src/main/java/org/jaggeryjs/hostobjects/xhr/XMLHttpRequestHostObject.java]: Fixed in af3ac874ea63a07e139fdd85b36eeeed3d15e410

**Review by 1akshitha — APPROVED** [review]

**Pull request merged**